### PR TITLE
Tests for natural and bound generators

### PR DESCRIPTION
**Pull request by robertlyson**
Hi,

I've added a couple of tests for our basic generators. 
If you can think of any other test cases I'll be happy to add them.

### CODE REVIEW

File: test/bound_natural_test.exs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  use ExUnitProperties
+  import Orangeade.Generator.BoundNatural, only: [stream: 1]
+
+  property "stream can't contain elements bigger than specified by limit" do
+    check all l <- integer(), l > 0 do
+      assert stream(limit: l) |> Caffeine.Stream.take(50) |> Enum.max() <= l
+    end
+  end
+
+  property "sum of generated elements is smaller or equal than multiplied limits" do
```
Comment (Dzol) [inline test/bound_natural_test.exs]: This is good. I only came up with one property that loosely resembles your first one though I gave it almost no time. Streams might be a simple example but the practice will help with property-based testing for more complex things in the future.

File: test/big_natural_test.exs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  use ExUnit.Case, async: true
+  use ExUnitProperties
+  import Orangeade.Generator.BigNatural, only: [stream: 0]
+
+  test "check 5 first generated elements" do
+    generated =
+      stream()
+      |> Caffeine.Stream.take(5)
+
+    assert generated == [
```
Comment (Dzol) [inline test/big_natural_test.exs]: Did you do the calculation manually?

File: test/big_natural_test.exs [hunk 3]
```diff
@@ -0,0 +1,19 @@
+defmodule Orangeade.Generator.BigNatural.Tests do
+  use ExUnit.Case, async: true
+  use ExUnitProperties
+  import Orangeade.Generator.BigNatural, only: [stream: 0]
+
+  test "check 5 first generated elements" do
+    generated =
+      stream()
+      |> Caffeine.Stream.take(5)
```
Comment (Dzol) [inline test/big_natural_test.exs]: I often say there's a time and place for piping... when it's something like the following:
```Elixir
alpha = 1..100
beta  = Enum.map(alpha, &FizzBuzz.stringify/1)
gamma = Enum.each(beta, &IO.puts/1)
```
Then the pipe is great:
```Elixir
1..100
|> Enum.map(&FizzBuzz.stringify/1)
|> Enum.each(&IO.puts/1)
```
On the other hand a lot of Elixir code I see pipes just once. The formatter doesn't seem to complain either way. You decide what you think is better style. I guess a fair comparison on style would be with method chaining in OO languages.

File: test/big_natural_test.exs [hunk 4]
```diff
@@ -0,0 +1,19 @@
+defmodule Orangeade.Generator.BigNatural.Tests do
+  use ExUnit.Case, async: true
+  use ExUnitProperties
```
Comment (Dzol) [inline test/big_natural_test.exs]: If we don't need this then we can remove it

**Review by Dzol — APPROVED** [review]
If there _are_ any changes you want to make following the comments then go for it in another PR 🙂 

If **you** have a chance this evening then merge it (it's nice to be the one to push the button) otherwise I'll do so first thing in the morning.

The test cases I had thought of were:
- `BigNatural` is always positive or zero
- `BoundNatural` is 0 ≤ x﹤limit

**Comment by Dzol** [comment]
Ah: I forgot!

Before you merge there were two changes...

1. Let's put the test modules in a directory like `test/orangeade/generator/*`. This is the convention and I believe editors and tooling rely on it to jump between implementation and test.

2. The module names for the test should have the suffix `Test` not `.Test`. I.e. w/o the period. It makes sense not to "dot" it because the test module isn't a child (conceptually) of the module it's testing. I guess this convention made more sense than putting all tests under a `Test` namespace.

File: test/big_natural_test.exs (hunk 4 above)
Comment (robertlyson) [reply test/big_natural_test.exs]: Good idea :+1:

File: test/big_natural_test.exs (hunk 3 above)
Comment (robertlyson) [reply test/big_natural_test.exs]: Thanks for the explanation, that makes sense :)

File: test/big_natural_test.exs (hunk 2 above)
Comment (robertlyson) [reply test/big_natural_test.exs]: No, rather a lazy way. I just assumed our implementation is correct .. what a mistake :/

**Comment by robertlyson** [comment]
@Dzol fixed. Thanks for all of the comments :+1:

File: test/big_natural_test.exs (hunk 2 above)
Comment (Dzol) [reply test/big_natural_test.exs]: This is a regression test then

**Pull request merged**